src/uiTest/resources/project.rhino.js

```diff
@@ -98,6 +98,23 @@ function Project(underlyingProject) {
     }
   };
 
+  this.getManagedBranchesAndCommits = function () {
+    const snapshot = getGraphTable().getGitMacheteRepositorySnapshot();
+    if (snapshot != null) {
+      return snapshot.getManagedBranches().flatMap(b => {
+        const branchesAndCommits = new LinkedList();
+        if (b.isNonRoot()) {
+          const nonRoot = b.asNonRoot();
+          branchesAndCommits.addAll(nonRoot.getUniqueCommits().map(c => c.getShortMessage()).toJavaList());
```

Comment:
it's here exactly - using `getUniqueCommits` works as expected (at least to me), but with `getCommitsUntilParent` the `Develop commit` is included in the result. 

Probably it's not from the develop branch (`isNonRoot` for develop works fine) but some other branch decided that the parent is root (not develop as it should) 🤔

Reply:
Wow okay.... weird 😯